server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferMetadata.java

```diff
@@ -49,6 +53,8 @@ public class TranslogTransferMetadata {
 
     private static final String METADATA_CODEC = "md";
 
+    public static final MetadataFilenameComparator METADATA_FILENAME_COMPARATOR = new MetadataFilenameComparator();
```

Comment:
Nitpick, but the type can be `Comparator<String>` so that they private type never leaves this class.